addons/delivery/models/delivery_grid.py

```diff
@@ -121,6 +121,7 @@ def _get_price_dict(self, total, weight, volume, quantity):
     def _get_price_from_picking(self, total, weight, volume, quantity):
         price = 0.0
         criteria_found = False
+        weight = self.env.context.get('order_weight') or weight
```

Comment:
I moved to `_get_price_available` but still needed a context since this function is also called in wizard onchange and the related field `shipping_weight` is not yet saved.